test/unit/sessions/test_session.py

```diff
@@ -2180,6 +2183,203 @@ def test_action_output_capture_filter_ja_upload_callback_valid(
         assert len(session._upload_manifest_list) == 1
         assert all(isinstance(item, UploadManifestInfo) for item in session._upload_manifest_list)
 
+    @pytest.mark.skipif(
+        not MANIFEST_REPORTING_FEATURE or not ASSET_SYNC_JOB_USER_FEATURE,
+        reason="This test only runs when both MANIFEST_REPORTING_FEATURE and ASSET_SYNC_JOB_USER_FEATURE are enabled",
+    )
```

Comment:
Rather than skipping the tests, does it make more sense to mock the feature flags while running this test instead of skip the tests depending on the feature flag values?

Otherwise we have to run the test suite multiple times with all feature flag combinations.

Same applies to other tests